### PR TITLE
refactor(ff): make site launch flag a bool flag

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Right now, our GGS flag adopts a two tiered approach - first, we get the list of repos from growthbook then we validate based on that. 

This is a _whitelist_ approach, where we add sites as they shift to ggs. This is effortful and going forward, we want to make our flag a _blacklist_ approach, where we allow sites by default until we need to blacklist them. 

## Solution
Add new flag on growthbook that returns a boolean value checking if the user type is email and whether site is blacklisted. (it's true if user is email && site isn't blacklisted)

## Deploy notes
**these steps have to be done pre deploy**
- [ ] for new **email login** sites that are not on ggs, migrate them to ggs by cloning them onto efs (can use form atm) 

**these steps have to be done post deploy**
- [ ] remove the flag for ggs whitelisted repo
- [ ] enable the flag for `is_ggs_enabled` on **production**

### CODE REVIEW

File: src/constants/featureFlags.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export const FEATURE_FLAGS = {
   GGS_WHITELISTED_REPOS: "ggs_whitelisted_repos",
```
Comment (kishore03109) [inline src/constants/featureFlags.ts:2]: Should we remove this since this is no longer in use?

File: src/middleware/routeHandler.js [hunk 2]
```diff
@@ -74,23 +73,15 @@ const attachWriteRouteHandlerWrapper = (routeHandler) => async (
...
-      FEATURE_FLAGS.GGS_WHITELISTED_REPOS,
-      {
-        repos: [],
-      }
-    )
-  }
-
   let isGitAvailable = true
+
   // only check git file lock if the repo is whitelisted
```
Comment (kishore03109) [inline src/middleware/routeHandler.js:78]: blacklisted**

File: src/services/db/RepoService.ts [hunk 3]
```diff
@@ -38,27 +38,19 @@ export default class RepoService extends GitHubService {
...
-    logger.info(
-      `Evaluating if ${repoName} is GGS whitelisted: ${ggsWhitelistedRepos.includes(
-        repoName
-      )}`
+    logger.info(`Evaluating if ${repoName} is GGS whitelisted.`)
+    // NOTE: Growthbook has to be optional
+    // as we cannot guarantee its presence on `sessionData`.
+    // It is present iff there is a `siteHandler` attached.
+    return !!growthbook?.getFeatureValue(
+      FEATURE_FLAGS.IS_GGS_WHITELISTED,
```
Comment (kishore03109) [inline src/services/db/RepoService.ts:51]: are we using the wrong feature fflag here?

**Review by kishore03109 — COMMENTED** [review]
How does this play with migrated sites? ie when is the git clone step for these sites going to occur ah?

File: src/constants/featureFlags.ts (hunk 1 above)
Comment (seaerchin) [reply src/constants/featureFlags.ts:2]: we cannot remove until this feature is released cos prod still uses this flag

File: src/middleware/routeHandler.js (hunk 2 above)
Comment (seaerchin) [reply src/middleware/routeHandler.js:78]: good catch, gonna update

File: src/services/db/RepoService.ts (hunk 3 above)
Comment (seaerchin) [reply src/services/db/RepoService.ts:51]: actually no, i just named it wrongly cos i think i got confused between white/blacklist

**Comment by seaerchin** [comment]
> How does this play with migrated sites? ie when is the git clone step for these sites going to occur ah?

there's a form for them to fill up that triggers a clone

**Comment by kishore03109** [comment]
@seaerchin wait so to be clear, post email migration of sites there needs to a manual sending submission of this form by eng/ops? 
cc. @alexanderleegs think we might need to take this into account ya

File: src/constants/featureFlags.ts (hunk 1 above)
Comment (harishv7) [reply src/constants/featureFlags.ts:2]: let's add a todo above

File: src/middleware/routeHandler.js [hunk 4]
```diff
@@ -25,14 +25,13 @@ const gitFileSystemService = new GitFileSystemService(
   new SimpleGit({ maxConcurrentProcesses: MAX_CONCURRENT_GIT_PROCESSES })
 )
 
-const isRepoWhitelisted = (siteName, ggsWhitelistedRepos) => {
+const isRepoWhitelisted = (siteName, growthbook) => {
```
Comment (harishv7) [inline src/middleware/routeHandler.js:28]: do we still need a function for this? seems like the sitename is only passed in to log it and we are just calling 1 growthbook function

File: src/constants/featureFlags.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
...
+  IS_GGS_WHITELISTED: "is_ggs_whitelisted",
```
Comment (harishv7) [inline src/constants/featureFlags.ts:3]: just to be clear are we doing a whitelist or blacklist?

File: src/constants/featureFlags.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
 export const FEATURE_FLAGS = {
```
Comment (harishv7) [inline src/constants/featureFlags.ts:1]: can we add deploy notes to the PR mentioning to turn on this flag for prod during release

File: src/middleware/routeHandler.js (hunk 4 above)
Comment (seaerchin) [reply src/middleware/routeHandler.js:28]: removed

File: src/constants/featureFlags.ts (hunk 5 above)
Comment (seaerchin) [reply src/constants/featureFlags.ts]: changed naming to `IS_GGS_ENABLED` for clarity

File: src/constants/featureFlags.ts (hunk 6 above)
Comment (seaerchin) [reply src/constants/featureFlags.ts:1]: done

File: src/services/db/RepoService.ts [hunk 7]
```diff
@@ -38,27 +38,19 @@ export default class RepoService extends GitHubService {
...
+  isRepoWhitelisted(
+    repoName: string,
+    growthbook?: GrowthBook<FeatureFlags>
+  ): boolean {
     // TODO: Adding for initial debugging if required. Remove once stabilised
-    logger.info(
-      `Evaluating if ${repoName} is GGS whitelisted: ${ggsWhitelistedRepos.includes(
-        repoName
-      )}`
+    logger.info(`Evaluating if ${repoName} is GGS whitelisted.`)
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/services/db/RepoService.ts]: function is removed

File: src/constants/featureFlags.ts (hunk 1 above)
Comment (seaerchin) [reply src/constants/featureFlags.ts]: removed in code, added deploy note to remove in gb

**Review by harishv7 — APPROVED** [review]

**Pull request merged**